### PR TITLE
Calendar integration ui

### PR DESCRIPTION
**Pull request by yanas**

### CODE REVIEW

File: react/features/base/react/components/NavigateSectionList.js [hunk 1]
```diff
@@ -93,7 +93,7 @@ class NavigateSectionList extends Component<Props> {
 
         return (
             <SectionList
-                ListEmptyComponent = { renderListEmptyComponent }
+                ListEmptyComponent = { renderListEmptyComponent() }
```
Comment (virtuacoplenny) [inline react/features/base/react/components/NavigateSectionList.js]: Was this change necessary? SectionList doc says ListEmptyComponent can be a render function.

File: react/features/base/react/components/web/NavigateSectionListItem.js [hunk 2]
```diff
@@ -6,17 +6,32 @@ import Container from './Container';
 import Text from './Text';
 import type { Item } from '../../Types';
 
-type Props = {
+export type Props = {
```
Comment (virtuacoplenny) [inline react/features/base/react/components/web/NavigateSectionListItem.js]: Is the export necessary? I may be missing where it's being used outside of this file.

File: react/features/welcome/components/WelcomePage.web.js [hunk 3]
```diff
@@ -211,6 +213,32 @@ class WelcomePage extends AbstractWelcomePage {
...
+     */
+    _renderTabs() {
+        const { t } = this.props;
+
+        const tabs = [
+            {
+                label: t('welcomepage.calendar'),
+                content: CalendarList ? <CalendarList /> : null,
+                defaultSelected: true
+            },
```
Comment (virtuacoplenny) [inline react/features/welcome/components/WelcomePage.web.js]: Atlaskit tabs automatically hides the tab if it has no content?

File: react/features/calendar-sync/components/index.js [hunk 4]
```diff
@@ -1,3 +1,3 @@
 export { default as ConferenceNotification } from './ConferenceNotification';
-export { default as MeetingList } from './MeetingList';
+export { default as CalendarList } from './CalendarList';
```
Comment (virtuacoplenny) [inline react/features/calendar-sync/components/index.js]: Gotta remove MeetingList.web.js.

File: react/features/calendar-sync/components/CalendarList.native.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
+// @flow
```
Comment (virtuacoplenny) [inline react/features/calendar-sync/components/CalendarList.native.js]: Was this file a copy/paste from MeetingList with no changes?

File: react/features/calendar-sync/components/CalendarList.native.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    /**
+     * Implements React's {@link Component#render}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { disabled } = this.props;
+
+        return (
+            AbstractCalendarList
```
Comment (virtuacoplenny) [inline react/features/calendar-sync/components/CalendarList.native.js]: Aren't abstract classes normally extended?

File: react/features/calendar-sync/components/AbstractCalendarList.js [hunk 7]
```diff
@@ -221,39 +182,56 @@ class MeetingList extends Component<Props> {
...
         const { createSection } = NavigateSectionList;
-        const nowSection = createSection(t('calendarSync.now'), 'now');
-        const nextSection = createSection(t('calendarSync.next'), 'next');
-        const laterSection = createSection(t('calendarSync.later'), 'later');
+        const TODAY_SECTION = 'today';
+        const todaySection
+            = createSection(t('calendarSync.today'), TODAY_SECTION);
+        const sectionMap = new Map();
+
+        sectionMap.set(TODAY_SECTION, todaySection);
```
Comment (virtuacoplenny) [inline react/features/calendar-sync/components/AbstractCalendarList.js]: I think this means sections will always have a length of at least 1, which means SectionList will never show ListEmptyComponent.

File: react/features/base/react/components/web/SectionList.js [hunk 8]
```diff
@@ -63,23 +70,25 @@ export default class SectionList extends Component<Props> {
...
-                            </Container>
-                        )
-                        )
+                                            return renderItem(listItem,
+                                                keyExtractor(section,
+                                                    listIndex));
+                                        }) }
+                                </Container>
+                            )
+                            )
```
Comment (virtuacoplenny) [inline react/features/base/react/components/web/SectionList.js]: A loading icon somewhere would be nice. It can take 2+ seconds for the calendar data to load.

File: react/features/base/react/components/web/NavigateSectionListItem.js [hunk 9]
```diff
@@ -68,6 +85,7 @@ export default class NavigateSectionListItem extends Component<Props> {
                     className = 'navigate-section-tile-body'>
                     { duration }
                 </Text>
```
Comment (virtuacoplenny) [inline react/features/base/react/components/web/NavigateSectionListItem.js]: Is NavigateSectionListItem specifically for calendar entries or do you know if it was intended to be something more general? It's markup makes it look like it was specific for calendar entries.

File: react/features/base/react/components/web/NavigateSectionListItem.js [hunk 10]
```diff
@@ -68,6 +85,7 @@ export default class NavigateSectionListItem extends Component<Props> {
                     className = 'navigate-section-tile-body'>
                     { duration }
```
Comment (virtuacoplenny) [inline react/features/base/react/components/web/NavigateSectionListItem.js]: Do you know much about the data here and why an array is used with assumption about what each index in the array stands for? It looks like index 0 might be meeting url but the variable name is date. Index 1 might be data but the variable name is duration.

File: react/features/base/react/components/NavigateSectionList.js [hunk 11]
```diff
@@ -93,7 +93,7 @@ class NavigateSectionList extends Component<Props> {
 
         return (
             <SectionList
-                ListEmptyComponent = { renderListEmptyComponent() }
```
Comment (yanas) [inline react/features/base/react/components/NavigateSectionList.js]: This was fixing an issue and was intentional. With the change below the case of disconnected calendar doesn't work anymore, so I reverted this change.

Comment (virtuacoplenny) [reply react/features/base/react/components/NavigateSectionList.js]: I have a fix I'm working on for it now, which should try to fix the empty component never displaying. It seems like it worked on native because it was calling _getRenderListEmptyComponent() in the calendarlist instead of passing in _getRenderListEmptyComponent.

File: react/features/base/react/components/web/NavigateSectionListItem.js [hunk 12]
```diff
@@ -73,19 +73,21 @@ export default class NavigateSectionListItem<P: Props>
...
+                    <Text
+                        className = 'navigate-section-tile-body'>
+                        { date }
+                    </Text>
+                    <Text
+                        className = 'navigate-section-tile-body'>
+                        { duration }
+                    </Text>
+                </Container>
+                { elementAfter || null }
```
Comment (yanas) [inline react/features/base/react/components/web/NavigateSectionListItem.js]: If we add the elementAfter, as a completely constructed element, then we don't need the action props I've added previously. I'm going to remove them.
